src/building/building.h

```diff
@@ -199,6 +199,7 @@ typedef struct building {
     unsigned char fumigation_direction;
     short resources[RESOURCE_MAX];
     unsigned char accepted_goods[RESOURCE_MAX];
+    unsigned char accept_delivery;
```

Comment:
Why not just use `accepted_goods[RESOURCE_WEAPONS]`?

That way you'd avoid the entire savegame hassle.